server/backend/cron/canteenHeadCount.go

```diff
@@ -0,0 +1,236 @@
+		{CanteenId: "stucafe-karlstr",
+			Target:   "",
+			MaxCount: 1000},
+		// Same as for the mensa-parsing since I can not distinguish them
+		{CanteenId: "stucafe-pasing",
+			Target:   "ap.apa{15,18}-?rl*.ssid.*",
+			MaxCount: 50},
+		// No data found
+		{CanteenId: "ipp-bistro",
+			Target:   "",
+			MaxCount: 1000},
+		// No data available since the RBG does not provide stats
+		{CanteenId: "fmi-bistro",
+			Target:   "",
+			MaxCount: 1000},
+		// No data found
+		{CanteenId: "mediziner-mensa",
+			Target:   "",
+			MaxCount: 1000},
+		// Don't know which APs are for the canteen: http://wlan.lrz.de/apstat/filter/Unterbezirk/ua/
+		{CanteenId: "mensa-straubing",
+			Target:   "",
+			MaxCount: 1000},
+	}
+)
+
+/*
+BASE_URL is the base URL for the required format.
+Contains the 'XXX' placeholder that has to replaced with the Target property of the
+CanteenApInformation when performing a request.
+*/
+const BASE_URL = "http://graphite-kom.srv.lrz.de/render/?from=-10min&target=XXX&format=json"
+
+func (c *CronService) canteenHeadCountCron() error {
+	log.Info("Updating canteen head count stats...")
+	for _, canteen := range CANTEENS {
+		if len(canteen.Target) <= 0 {
+			log.Debug("Skipping canteen head count stats for '", canteen.CanteenId, "', since there is no target.")
+			continue
+		}
+
+		log.Debug("Updating canteen head count stats for: ", canteen.CanteenId)
+		aps := requestApData(&canteen)
+		if len(aps) <= 0 {
+			log.Debug("No canteen head count data points found for: ", canteen.CanteenId)
+			continue
+		}
+
+		count := sumApCounts(aps)
+		updateDb(&canteen, count, c.db)
+		log.Debug("Canteen head count stats (", count, ") updated for: ", canteen.CanteenId)
+	}
+	log.Info("Canteen head count stats updated.")
+	return nil
+}
+
+func sumApCounts(aps []AccessPoint) uint32 {
+	total := uint32(0)
+	for _, ap := range aps {
+		_, count, err := parseAp(&ap)
+		if err != nil {
+			log.WithError(err).Error("Canteen HeadCount getting the count failed for access point: ", ap.Target)
+			continue
+		}
+		total += uint32(count)
+	}
+	return total
+}
+
+func updateDb(canteen *CanteenApInformation, count uint32, db *gorm.DB) error {
+	entry := model.CanteenHeadCount{
+		CanteenId: canteen.CanteenId,
+		Count:     count,
+		MaxCount:  canteen.MaxCount,
+		Percent:   (float32(count) / float32(canteen.MaxCount)) * 100,
+		Timestamp: time.Now(),
+	}
+
+	res := db.Model(&model.CanteenHeadCount{}).Where(model.CanteenHeadCount{CanteenId: canteen.CanteenId}).Updates(&entry)
+	if res.Error != nil {
+		return res.Error
+	}
+
+	if res.RowsAffected == 0 {
+		return db.Create(&entry).Error
+	}
+	return nil
+}
+
+func requestApData(canteen *CanteenApInformation) []AccessPoint {
+	// Perform web request
+	url := strings.Replace(BASE_URL, "XXX", canteen.Target, 1)
+	resp, err := http.Get(url)
+	if err != nil {
+		log.WithError(err).Error("Canteen HeadCount web request failed for: ", canteen.CanteenId)
+		return []AccessPoint{}
```

Comment:
Would be possible but in my eyes it clutters the for-each loop too much where I'm calling it from. Bun in case you insist on it I can change the signature for it.